lib/messaging/message_stream.dart

```diff
@@ -0,0 +1,122 @@
+import 'dart:async';
+
+import 'package:covidactnow/messaging/message.dart';
+import 'package:covidactnow/messaging/message_dialog.dart';
+import 'package:covidactnow/utils/map_ext.dart';
+import 'package:firebase_messaging/firebase_messaging.dart';
+import 'package:flutter/material.dart';
+
+/*
+
+  To send a message, go to the firebase Cloud messaging panel
+
+  Additional options:
+    Custom data:
+      click_action: FLUTTER_NOTIFICATION_CLICK
+      route: <State abbr>
+
+*/
+
+class MessageStream {
+  factory MessageStream() {
+    _singleton ??= MessageStream._();
+
+    return _singleton;
+  }
+
+  MessageStream._() {
+    _setup();
+  }
+
+  static MessageStream _singleton;
+  FirebaseMessaging _firebaseMessaging;
+  StreamController<Map<String, dynamic>> _streamController;
+
+  Stream<Map> get stream => _streamController.stream;
+  Future<String> get token => _firebaseMessaging.getToken();
+
+  // called from main.dart to listen for notifications
+  static void listen(BuildContext context) {
+    final MessageStream _messageStream = MessageStream();
+
+    _messageStream.stream.forEach((Map event) {
+      print(event.toString());
+
+      switch (event.strForKey('event')) {
+        case 'onMessage':
```

Comment:
Maybe let's make it its own class instead of encoding the event types as strings?

Reply:
done.